src/PlaneWaveBasis.jl

```diff
@@ -140,24 +150,17 @@ end
 build_kpoints(basis::PlaneWaveBasis, kcoords) =
     build_kpoints(basis.model, basis.fft_size, kcoords, basis.Ecut)
 
-# This is the "internal" constructor; the higher-level one below should be preferred
+# Lowest-level constructor. All given parameters must be the same on all processors
+# and are stored in PlaneWaveBasis for easy reconstruction
 @timing function PlaneWaveBasis(model::Model{T}, Ecut::Number,
-                                kcoords::AbstractVector, ksymops, symmetries=nothing;
-                                fft_size=nothing, variational=true,
-                                optimize_fft_size=false, supersampling=2,
-                                kgrid=nothing, kshift=nothing,
-                                comm_kpts=MPI.COMM_WORLD,
-                               ) where {T <: Real}
-    mpi_ensure_initialized()
-
-    # Validate kpoints and symmetries
-    @assert length(kcoords) == length(ksymops)
-    if symmetries === nothing
-        # TODO instead compute the group generated by with ksymops, or
-        # just retire this constructor. Not critical because this
-        # should not be used in this context anyway...
-        symmetries = vcat(ksymops...)
+                                kcoords::AbstractVector, ksymops,
+                                fft_size, variational, symmetries,
+                                kgrid, kshift, comm_kpts) where {T <: Real}
+    if !(all(fft_size .== next_working_fft_size.(T, fft_size)))
+        error("Selected fft_size will not work for the buggy generic FFT routines")
     end
+    fft_size = Tuple{Int, Int, Int}(fft_size) # need explicit convert from 
```

Comment:
two spaces before `#` (is it that hard to write an emacs macro that fixes that?)

Reply:
my avowed goal is that at some point you'll give up :-p I still don't understand why you care about that, I've never seen that rule before and it makes no sense to me

Reply:
it makes comments easier to read. python uses the same convention (and so do the established julia styles if i remember correctly)